src/app/organisms/noti-list/NotiList.tsx

```diff
@@ -40,10 +40,40 @@ const renderBody = (body: string, customBody: string | undefined) => {
 export function useNotifications() {
   const [notis, setNotis] = useState<any[]>([]);
   const [nextToken, setNextToken] = useState(null);
+  const [isLoading, setLoading] = useState(false);
   const [receivedNotifications, setReceivedNotifications] = useState(0);
 
   const mx = initMatrix.matrixClient;
 
+  const fetchNotifications = async (from?:any, limit = 10) => {
+    if (!mx) return;
+
+    const url = new URL('/_matrix/client/v3/notifications', mx.getHomeserverUrl());
+    const params = new URLSearchParams();
+    
+    params.append('limit', limit.toString());
+    if (from != null) params.append('from', from.toString());
+
+    url.search = params.toString();
+
+    const notificationsResponse = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${mx.getAccessToken()}`,
+      },
+    }).then((res) => res.json());
+
+    const { notifications, next_token } = notificationsResponse;
+
+    for (const noti of notifications) {
+        if (notis.find(x => x.event.event_id == noti.event.event_id) != null) continue;
+
+        setNotis(notis => [...notis, noti].sort((a, b) => b.ts - a.ts));
```

Comment:
let's do that only a single time instead of in a loop, since we can just use

```
setNotis(notis => [...notis, ...notifications].sort((a, b) => b.ts - a.ts));
```

below

Reply:
this duplicates notifications since it's ran multiple times

Reply:
ah right, but I think it'd be better to move the filter condition into the callback here anyway, since then we are guaranteed to have the latest `notis` value and won't run into race conditions with stuff like instantly finished service workers or slow browsers or other edge cases.

Reply:
```ts
const newNotis:any[] = [];

for (const noti of notifications) {
    if (notis.find(x => x.event.event_id == noti.event.event_id) != null) continue;
    newNotis.push(noti);
}

setNotis(notis => [...notis, ...newNotis].sort((a, b) => b.ts - a.ts));
```

You mean like this?